policies/policies.go

```diff
@@ -0,0 +1,301 @@
+				logger.Errorf("Error creating repo file: %v", err)
+				errs = append(errs, fmt.Sprintf("error creating googet repo file: %v", err))
+			}
+		}
+		if err := googetRepositories(gooRepos, config.GooGetRepoFilePath()); err != nil {
+			logger.Errorf("Error writing googet repo file: %v", err)
+			errs = append(errs, fmt.Sprintf("error writing googet repo file: %v", err))
+		}
+		if err := googetChanges(gooInstalled, gooRemoved, gooUpdated); err != nil {
+			errs = append(errs, fmt.Sprintf("error performing googet changes: %v", err))
+		}
+	}
+
+	if packages.AptExists {
+		if _, err := os.Stat(config.AptRepoFilePath()); os.IsNotExist(err) {
+			logger.Debugf("Repo file does not exist, will create one...")
+			if err := os.MkdirAll(filepath.Dir(config.AptRepoFilePath()), 07550); err != nil {
+				logger.Errorf("Error creating repo file: %v", err)
+				errs = append(errs, fmt.Sprintf("error creating apt repo file: %v", err))
+			}
+		}
+		if err := aptRepositories(aptRepos, config.AptRepoFilePath()); err != nil {
+			logger.Errorf("Error writing apt repo file: %v", err)
+			errs = append(errs, fmt.Sprintf("error writing apt repo file: %v", err))
+		}
+		if err := aptChanges(aptInstalled, aptRemoved, aptUpdated); err != nil {
+			errs = append(errs, fmt.Sprintf("error performing apt changes: %v", err))
+		}
+	}
+
+	if packages.YumExists {
+		if _, err := os.Stat(config.YumRepoFilePath()); os.IsNotExist(err) {
+			logger.Debugf("Repo file does not exist, will create one...")
+			if err := os.MkdirAll(filepath.Dir(config.YumRepoFilePath()), 07550); err != nil {
+				logger.Errorf("Error creating repo file: %v", err)
+				errs = append(errs, fmt.Sprintf("error creating yum repo file: %v", err))
+			}
+		}
+		if err := yumRepositories(yumRepos, config.YumRepoFilePath()); err != nil {
+			logger.Errorf("Error writing yum repo file: %v", err)
+			errs = append(errs, fmt.Sprintf("error writing yum repo file: %v", err))
+		}
+		if err := yumChanges(yumInstalled, yumRemoved, yumUpdated); err != nil {
+			errs = append(errs, fmt.Sprintf("error performing yum changes: %v", err))
+		}
+	}
+
+	if packages.ZypperExists {
+		if _, err := os.Stat(config.ZypperRepoFilePath()); os.IsNotExist(err) {
+			logger.Debugf("Repo file does not exist, will create one...")
+			if err := os.MkdirAll(filepath.Dir(config.ZypperRepoFilePath()), 07550); err != nil {
+				logger.Errorf("Error creating repo file: %v", err)
+				errs = append(errs, fmt.Sprintf("error creating zypper repo file: %v", err))
+			}
+		}
+		if err := zypperRepositories(zypperRepos, config.ZypperRepoFilePath()); err != nil {
+			logger.Errorf("Error writing zypper repo file: %v", err)
+			errs = append(errs, fmt.Sprintf("error writing zypper repo file: %v", err))
+		}
+		if err := zypperChanges(zypperInstalled, zypperRemoved, zypperUpdated); err != nil {
+			errs = append(errs, fmt.Sprintf("error performing zypper changes: %v", err))
```

Comment:
Factoring this out into package manager based interface methods may help remove the duplication here too.